core/src/main/java/io/questdb/VolumeDefinitions.java

```diff
@@ -0,0 +1,139 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2022 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb;
+
+import io.questdb.std.*;
+import io.questdb.std.str.Path;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class VolumeDefinitions {
+
+    private static final char SEPARATOR = ',';
+
+    private final LowerCaseCharSequenceObjHashMap<String> aliasToVolumeRoot = new LowerCaseCharSequenceObjHashMap<>(4);
+    private int hi;
+    private int limit;
+    private int lo;
+
+    public void forEach(LowerCaseCharSequenceObjHashMap.CharSequenceObjConsumer<String> action) {
+        aliasToVolumeRoot.forEach(action);
+    }
+
+    public VolumeDefinitions of(@Nullable CharSequence definitions, @NotNull Path path, @NotNull String root) throws ServerConfigurationException {
+        if (definitions != null) {
+            // 'any-case-alias' -> 'absolute path to volume' (quotes are optional)
```

Comment:
Given this Dockerfile:

```Dockerfile
#
#     ___                  _   ____  ____
#    / _ \ _   _  ___  ___| |_|  _ \| __ )
#   | | | | | | |/ _ \/ __| __| | | |  _ \
#   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
#    \__\_\\__,_|\___||___/\__|____/|____/
#
#  Copyright (c) 2014-2019 Appsicle
#  Copyright (c) 2019-2023 QuestDB
#
#  Licensed under the Apache License, Version 2.0 (the "License");
#  you may not use this file except in compliance with the License.
#  You may obtain a copy of the License at
#
#  http://www.apache.org/licenses/LICENSE-2.0
#
#  Unless required by applicable law or agreed to in writing, software
#  distributed under the License is distributed on an "AS IS" BASIS,
#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
#  See the License for the specific language governing permissions and
#  limitations under the License.
#

FROM python:3.10.9-slim-buster

EXPOSE 8888/tcp
EXPOSE 8812/tcp
EXPOSE 9000/tcp
EXPOSE 9009/tcp

ENV QUESTDB_TAG=6.7
ENV ARCHITECTURE=x64
ENV PYTHONUNBUFFERED 1
ENV JAVA_HOME=/usr/lib/jvm/java-17-amazon-corretto
ENV PATH="$JAVA_HOME/bin:${PATH}"

# Update system
RUN apt-get -y update
RUN apt-get -y upgrade
RUN apt-get -y --no-install-recommends install syslog-ng ca-certificates git curl wget vim procps gnupg2 lsb-release software-properties-common unzip less tar gzip iputils-ping

# Install JDK
RUN wget -O- https://apt.corretto.aws/corretto.key | gpg --dearmor | tee /etc/apt/trusted.gpg.d/winehq.gpg >/dev/null && \
    add-apt-repository 'deb https://apt.corretto.aws stable main' && \
    apt-get update && \
    apt-get install -y java-17-amazon-corretto-jdk=1:17.0.3.6-1

# Clean after packages installation
RUN apt-get clean
RUN rm -rf /var/lib/apt/lists/*

# Aliases
RUN echo "alias l='ls -l'" >> ~/.bashrc
RUN echo "alias ll='ls -la'" >> ~/.bashrc
RUN echo "alias rm='rm -i'" >> ~/.bashrc

WORKDIR /opt

# Install QuestDB
COPY questdb-6.7.1-SNAPSHOT-no-jre-bin.tar.gz questdb.tar.gz
RUN tar xvfz questdb.tar.gz
RUN rm questdb.tar.gz
RUN mv questdb-6.7.1-SNAPSHOT-no-jre-bin questdb

# Configure QuestDB
RUN ulimit -S unlimited
RUN ulimit -H unlimited
RUN mkdir csv
RUN mkdir tmp
RUN mkdir backups
RUN mkdir volume0
RUN mkdir volume1
RUN mkdir questdb/db
RUN mkdir questdb/conf
RUN echo "config.validation.strict=true" > questdb/conf/server.conf
RUN echo "query.timeout.sec=120" >> questdb/conf/server.conf
RUN echo "cairo.volumes=volume0->/opt/volume0, volume1->/opt/volume1" >> questdb/conf/server.conf
RUN echo "cairo.sql.copy.root=/opt/csv" >> questdb/conf/server.conf
RUN echo "cairo.sql.copy.work.root=/opt/tmp" >> questdb/conf/server.conf
RUN echo "cairo.sql.backup.root=/opt/backups" >> questdb/conf/server.conf

# Install requirements.txt
COPY requirements.txt .
RUN pip install --upgrade pip
RUN pip install --no-compile --only-binary :all: -r requirements.txt
RUN jupyter-lab --generate-config && sed -i -e "s|# c.ServerApp.allow_remote_access = False|# c.ServerApp.allow_remote_access = True|g" /root/.jupyter/jupyter_lab_config.py

# Create run.sh script
RUN echo "#!/bin/bash" > run.sh
RUN echo "/opt/questdb/questdb.sh start -d /opt/questdb" >> run.sh
RUN echo "jupyter-lab --allow-root --ip 0.0.0.0 --port 8888 --no-browser --notebook-dir /opt/notebooks /opt/notebooks/play.ipynb" >> run.sh
RUN chmod 700 run.sh

COPY notebooks notebooks
CMD ["/bin/bash", "-c", "/opt/run.sh"]
```

(you would need to build `questdb-6.7.1-SNAPSHOT-no-jre-bin.tar.gz questdb.tar.gz` from my branch and place it alongside the other ^ ^ Dockerfile)

(you would build the image: `docker build -t io.questdb.play:1.0-SNAPSHOT .`)

This command will work just fine:

```shell
docker run --rm \
    -p 8888:8888 \
    -p 8812:8812 \
    -p 9009:9009 \
    -p 9000:9000 \
    --name questdb-play \
    -v /Users/marregui/QUEST/db:/opt/questdb/db \
    -v /Users/marregui/QUEST/notebooks:/opt/notebooks \
    -v /Users/marregui/QUEST/backups:/opt/backups \
    -v /Users/marregui/QUEST/csv:/opt/csv \
    -v /Users/marregui/OTHER/volume0:/opt/volume0 \
    -v /Users/marregui/OTHER/volume1:/opt/volume1 \
    -it io.questdb.play:1.0-SNAPSHOT
```

segregating the QuestDB file system at that level of granularity.

Notice the `cairo.volumes=volume0->/opt/volume0, volume1->/opt/volume1"` in **server.conf** .

Then these queries will work from the web console:

```sql
create table if not exists trade1 (
    sym symbol index capacity 128,
    px long,
    qty int,
    leverage float,
    ts timestamp
) timestamp(ts) partition by day in volume volume0;

insert into trade1 values('A', 200000, 3, 0.9,  '2022-12-31T12:00:00.000000Z');
insert into trade1 values('B', 198000, 4, 0.97, '2022-12-31T12:00:10.000000Z');
insert into trade1 values('C', 199000, 4, 0.85, '2022-12-31T12:00:11.000000Z');
insert into trade1 values('A', 200000, 3, 0.9,  '2023-01-01T12:00:00.000000Z');
insert into trade1 values('B', 198000, 4, 0.97, '2023-01-01T12:00:10.000000Z');
insert into trade1 values('C', 199000, 4, 0.85, '2023-01-01T12:00:11.000000Z');

create table if not exists byb_ltinfo (
    sym symbol index capacity 128,
    price long,
    quantity int,
    ts timestamp
) timestamp(ts) partition by day in volume volume1;


insert into byb_ltinfo values('A', 200300, 2, '2022-12-31T12:00:01.000000Z');
insert into byb_ltinfo values('B', 198200, 1, '2022-12-31T12:00:11.000000Z');
insert into byb_ltinfo values('C', 199000, 1, '2022-12-31T12:00:12.000000Z');
insert into byb_ltinfo values('A', 200300, 2, '2023-01-01T12:00:01.000000Z');
insert into byb_ltinfo values('B', 198200, 1, '2023-01-01T12:00:11.000000Z');
insert into byb_ltinfo values('C', 199000, 1, '2023-01-01T12:00:12.000000Z');
```

each table will be in a different volume.

Reply:
Is the only way to make it work in docker is to rebuild with custom Dockerfile? Can it work with the docker image we officially build?

Can you sum this up in PR description please? Once I resolve this comment it will all go hidden.